packages/nextjs/pages/index.tsx

```diff
@@ -2,8 +2,16 @@ import Link from "next/link";
 import type { NextPage } from "next";
 import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
 import { MetaHeader } from "~~/components/MetaHeader";
+import { useScaffoldContractRead } from "~~/hooks/scaffold-eth/useScaffoldContractRead";
 
 const Home: NextPage = () => {
+  const { data: checkedInCount } = useScaffoldContractRead({
+    contractName: "BatchRegistry",
+    functionName: "checkedInCounter",
+    watch: true,
+  });
+  const loadingSpinner = <span className="loading loading-spinner loading-xs" />;
```

Comment:
We have a custom `<Spinner />` component that you could use here too.

We should probably update the component in SE-2, since we can simplify it with Daisy classes.
cc @technophile-04